### DISCUSSION TITLE
Issue binding the text property of a TextBlock with List property and index from Carousel

### DISCUSSION THREAD

**Discussion by Ghisso** [AvaloniaUI/Avalonia#9348] [Q&A] [answered]
I have an issue binding the text property of a TextBlock to a list property but with the index from the SelectedIndex of a carousel.
Basically, I am using a Carousel named "carousel" to display images, my ViewModel has the property Images, and I have successfully bound the carousel to the Images list and they display perfectly.
However, I would like to have a TextBlock which shows the name of the file associated to the images. I have a FileNames property in my ViewModel which is a List<string>.
But when I do this:

`<TextBlock Name="TxtBlocFilename" Text="{Binding FileNames[#carousel.SelectedIndex]}" HorizontalAlignment="Center"></TextBlock>`

nothing appears.

To check, I tried `Text="{Binding FileNames[0]}"` and `Text="{Binding #carousel.SelectedIndex}"`
and both work no problem, but when combing the 2 nothing appears.

Basically, I want to bind the text to my property but also using the SelectedIndex from the carousel at the same time to get the corresponding file name.
Could you please tell what I did wrong and how to fix my issue?

**Comment by FoggyFinder** [comment, marked as answer]
Not sure Avalonia support this scenario. Maybe it will be easier to bind to `SelectedItem`?

`#carousel.SelectedItem`

Or 

```xml
<TextBlock Name="TxtBlocFilename" Text="{Binding #carousel.SelectedItem.ImagePath}"></TextBlock>
```

where ImagePath is property of your object that describes image.

**Reply by timunie** [reply]
@FoggyFinder If this comment helped you, please mark it as the answer. Thx.

**Reply by FoggyFinder** [reply]
you probably meant @Ghisso ? My answer may be outdated though.

**Reply by timunie** [reply]
ah sorry yes 👍

**Comment by timunie** [comment]
Converted this into Q&A as this is neither a bug nor a feature request. If you disagree please let us know.
